### PR TITLE
Add mriGdbConsole_WriteSizeString and mriBuffer_WriteSizedStringAsHex

### PR DESCRIPTION
**Pull request by xiaoxiang781216**

### CODE REVIEW

**Comment by adamgreen** [comment]
Initial code review looks good. I like this change.  I will test and merge later this week.

File: core/gdb_console.c [hunk 1]
```diff
@@ -21,23 +21,22 @@
 #include <core/gdb_console.h>
 
 
-static void writeStringToExclusiveGdbCommChannel(const char* pString);
 void WriteStringToGdbConsole(const char* pString)
 {
-    writeStringToExclusiveGdbCommChannel(pString);
+    WriteSizeStringToGdbConsole(pString, mri_strlen(pString));
```
Comment (adamgreen) [inline core/gdb_console.c:26]: Doesn't compile! **gdb_console.c** is missing a `#include <core/libc.h>` to declare mri_strlen().

File: core/gdb_console.c [hunk 2]
```diff
@@ -21,23 +21,22 @@
...
 }
 
 /* Send the 'O' command to gdb to output text to its console.
 
     Command Format: OXX...
     Where XX is the hexadecimal representation of each character in the string to be sent to the gdb console.
 */
-static void writeStringToExclusiveGdbCommChannel(const char* pString)
+void WriteSizeStringToGdbConsole(const char* pString, size_t length)
```
Comment (adamgreen) [inline core/gdb_console.c:34]: Should be called `WriteSizedStringToGdbConsole()` WriteSized vs WriteSize

File: core/gdb_console.c [hunk 3]
```diff
@@ -21,23 +21,22 @@
...
 {
     Buffer* pBuffer = GetInitializedBuffer();
 
     Buffer_WriteChar(pBuffer, 'O');
-    Buffer_WriteStringAsHex(pBuffer, pString);
+    Buffer_WriteSizeStringAsHex(pBuffer, pString, length);
```
Comment (adamgreen) [inline core/gdb_console.c:39]: Fails to compile this line as well. Should be Buffer_WriteSize**d**StringAsHex().

**Comment by adamgreen** [comment]
Needs a few fixes. Once those are made, I will pull down again and start testing.

File: core/gdb_console.c (hunk 1 above)
Comment (xiaoxiang781216) [reply core/gdb_console.c]: Done.

File: core/gdb_console.c (hunk 2 above)
Comment (xiaoxiang781216) [reply core/gdb_console.c]: Done.

**Pull request merged**